### PR TITLE
feat:Support for BNC and vDOT assets

### PR DESCRIPTION
**Pull request by hqwangningbo**
**Pull Request Summary**

> Support for BNC and vDOT assets

### CODE REVIEW

**Comment by impelcrypto** [comment]
Hi @hqwangningbo 
Thank you for your contribution. Kindly submit required subscan links and staging URL for testing. Thanks.
https://docs.astar.network/docs/xcm/integration/add-to-portal#requirements-for-creating-a-pr

**Comment by impelcrypto** [comment]
Thanks, could you deploy this branch to somewhere like Vercel? So that our team can test it.
https://docs.astar.network/docs/builder-guides/Integration%20and%20Toolings/deploy-astar-portal

**Comment by impelcrypto** [comment]
Thanks, could you deploy this branch to somewhere like Vercel? So that our team can test it.
https://docs.astar.network/docs/builder-guides/Integration%20and%20Toolings/deploy-astar-portal

**Comment by hqwangningbo** [comment]
Our front end has just been successfully deployed, and the test link is here. 
https://astar.test.liebi.com/#/astar/assets

**Comment by impelcrypto** [comment]
thanks, could you send some BNC and vDOT to 5H6DC1YjyD531FiGxPCpWY3WNJrp5kQ35xHm6wibKuFvWzKW for testing? CC: @fiexer @bobo-k2 @sirius651 @Kahonnohak

**Comment by hqwangningbo** [comment]
@impelcrypto @Kahonnohak Some related tokens have been transferred

File: src/v2/repositories/implementations/xcm/BifrostXcmRepository.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  public async getTokenBalance(
+    address: string,
+    chain: XcmChain,
+    token: Asset,
+    isNativeToken: boolean
+  ): Promise<string> {
+    const api = await this.apiFactory.get(chain.endpoint);
+    try {
+      if (token.originAssetId == 'BNC') {
+        console.log(isNativeToken);
```
Comment (impelcrypto) [inline src/v2/repositories/implementations/xcm/BifrostXcmRepository.ts:87]: Let's remove this

File: src/v2/repositories/implementations/xcm/BifrostXcmRepository.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        return (await this.getNativeBalance(address, chain)).toString();
+      } else if (token.originAssetId == '0') {
```
Comment (impelcrypto) [inline src/v2/repositories/implementations/xcm/BifrostXcmRepository.ts:89]: what is '0'? Please make a const or add some comments.

File: src/v2/repositories/implementations/xcm/BifrostXcmRepository.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        const bal = await api.query.tokens.accounts<TokensAccounts>(address, {
+          VToken2: token.originAssetId,
+        });
+        return bal.free.toString();
+      } else {
+        const bal = await api.query.tokens.accounts<TokensAccounts>(address, {
+          Token2: 3,
```
Comment (impelcrypto) [inline src/v2/repositories/implementations/xcm/BifrostXcmRepository.ts:96]: maybe `this.astarTokens.ASTR`?

**Review by impelcrypto — CHANGES_REQUESTED** [review]

File: src/v2/repositories/implementations/xcm/BifrostXcmRepository.ts (hunk 2 above)
Comment (hqwangningbo) [reply src/v2/repositories/implementations/xcm/BifrostXcmRepository.ts:89]: vToken2: 0 represents vDOT on our chain

File: src/v2/repositories/implementations/xcm/BifrostXcmRepository.ts (hunk 3 above)
Comment (hqwangningbo) [reply src/v2/repositories/implementations/xcm/BifrostXcmRepository.ts:96]: Token2: 3 represents ASTA in our chain

File: src/v2/repositories/implementations/xcm/BifrostXcmRepository.ts (hunk 2 above)
Comment (bobo-k2) [reply src/v2/repositories/implementations/xcm/BifrostXcmRepository.ts:88]: Please to use constant here to define vDOT. Do the same for ASTR value on your chain

**Review by bobo-k2 — CHANGES_REQUESTED** [review]

**Comment by hqwangningbo** [comment]
@bobo-k2 @impelcrypto I have edited

**Comment by Kahonnohak** [comment]
Deposit and Withdraw to/from Astar Native/EVM were all successful!

**Comment by fiexer** [comment]
Could we also add Shiden support in this PR?
We also have open channels on Kusama, would be great to include this in the release.

**Review by impelcrypto — APPROVED** [review]
LGTM

**Review by bobo-k2 — APPROVED** [review]
LGTM

**Pull request merged**